src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Exports/AssemblyLoadContextWrapper.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#if !NET472
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Loader;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Exports;
+
+internal sealed class AssemblyLoadContextWrapper
+{
+    private static readonly ConcurrentDictionary<AssemblyName, Assembly?> s_loadedSharedAssemblies = new(AssemblyNameComparer.Default);
+
+    private AssemblyLoadContext? _assemblyLoadContext;
+    private readonly ImmutableDictionary<string, Assembly> _loadedAssemblies;
+
+    private AssemblyLoadContextWrapper(AssemblyLoadContext assemblyLoadContext, ImmutableDictionary<string, Assembly> loadedFiles)
+    {
+        _assemblyLoadContext = assemblyLoadContext;
+        _loadedAssemblies = loadedFiles;
+    }
+
+    public static bool TryLoadExtension(string assemblyFilePath, string? sharedDependenciesPath, [NotNullWhen(true)] out Assembly? assembly)
+    {
+        var dir = Path.GetDirectoryName(assemblyFilePath);
+        var fileName = Path.GetFileName(assemblyFilePath);
+        var fileNameNoExt = Path.GetFileNameWithoutExtension(assemblyFilePath);
+
+        var loadContext = TryCreate(fileNameNoExt, dir!, sharedDependenciesPath);
+        if (loadContext != null)
+        {
+            assembly = loadContext.GetAssembly(fileName);
+            return true;
+        }
+
+        assembly = null;
+        return false;
+    }
+
+    public static AssemblyLoadContextWrapper? TryCreate(string name, string assembliesDirectoryPath, string? sharedDependenciesPath)
+    {
+        try
+        {
+            var loadContext = CreateLoadContext(name, sharedDependenciesPath);
+            var directory = new DirectoryInfo(assembliesDirectoryPath);
+            var builder = new Dictionary<string, Assembly>();
+            foreach (var file in directory.GetFiles("*.dll"))
+            {
+                // HACK: For some reason trying to load ProjectEngineHost here causes a second version of the
```

Comment:
If we don't need the ProjectEngineHost then we can avoid publishing it to the telemetry package right?

Reply:
Refactored this bit, so the code above is outdated. We don't need this hack anymore